### DISCUSSION TITLE
Azure AD Authentication - Can we make the Login property variable set via claim?

### DISCUSSION THREAD

**Discussion by Breee** [grafana/grafana#51712] [Platform]
When using Azure AD authentication,  currently the login is set to the email-address of the user: 

https://github.com/grafana/grafana/blob/d2ab3556fa0babba43598a303e2f9f37970c7172/pkg/login/social/azuread_oauth.go#L93

I think it would be useful to set the Login,  using a claim provided by Azure AD.  
Because email-addresses sometimes can change in our company and we would like to use something that is truly unique.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
